### DISCUSSION TITLE
Permission error when running pip install, and again when running collectstatic script

### DISCUSSION THREAD

**Discussion by vapecastle** [netbox-community/netbox#15666] [Help Wanted!]
Hi all,

I've installed the latest Netbox and used workarounds to install the floorplan plugin, then restart services as per the plugin install guide.

In netbox I can't see the floorplan tab under my test location, which indicates that the plugin isn't loading correctly.

The plugin is however showing in netbox admin GUI as being installed, and the version is 0.3.6 which looks good.

I suspect I have set the user permissions up wrong, or one of the processes netbox uses isn't able to access this plugin, or my workarounds have caused issue.

**Permission error during pip install:**

> (venv) azureuser@Netbox-NBL:/opt/netbox/netbox$ pip3 install netbox-floorplan-plugin
> Defaulting to user installation because normal site-packages is not writeable

^Workaround was to manually move the 2x folders that pip3 downloaded (netbox_floorplan and netbox_floorplan_plugin-0.3.6.dist-info) into /opt/netbox-3.7.4/venv/lib/python3.10/site-packages

i.e:

sudo mv /usr/local/lib/python3.10/dist-packages/netbox_floorplan /opt/netbox/venv/lib/python3.10/site-packages/
sudo mv /usr/local/lib/python3.10/dist-packages/netbox_floorplan_plugin-0.3.6.dist-info /opt/netbox/venv/lib/python3.10/site-packages/

**Permission error during collectstatic script:**

> azureuser@Netbox-NBL:/opt/netbox/netbox$ source /opt/netbox/venv/bin/activate
> (venv) azureuser@Netbox-NBL:/opt/netbox/netbox$ python3 manage.py collectstatic
> 
> You have requested to collect static files at the destination
> location as specified in your settings:
> 
> /opt/netbox-3.7.4/netbox/static
> 
> This will overwrite existing files!
> Are you sure you want to do this?
> 
> Type 'yes' to continue, or 'no' to cancel: yes
> Traceback (most recent call last):
> 
> [omitted]
> 
> mkdir(name, mode)
> PermissionError: [Errno 13] Permission denied: '/opt/netbox-3.7.4/netbox/static/netbox_floorplan'

The workaround was to run the below command, which worked.

sudo /opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py collectstatic

The output was that 5 files were migrated, and I checked and they were actually there in the static folder.

Then I restarted netbox as per the plugins documentation, but no success.

I'm quite new to virtual environments and also pip so the answer here is probably pretty basic. Any help is greatly appreciated.

Cheers,

Hamish

**Comment by markkuleinio** [comment]
> (venv) azureuser@Netbox-NBL:/opt/netbox/netbox$ pip3 install netbox-floorplan-plugin

`sudo /opt/netbox/venv/bin/pip install netbox-floorplan-plugin`

(No need to separately activate the venv)

**Reply by markkuleinio** [reply]
But I don't have any experience of that specific plugin so I don't know how it should work in general

**Comment by Mylevsky** [comment]
```
sudo -s
cd /opt/netbox/netbox/
source /opt/netbox/venv/bin/activate
python3 manage.py migrate
sudo systemctl restart netbox
```

**Reply by vapecastle** [reply]
Thanks, this worked for me.
